generator/config/query/bitsAllClear.yaml

```diff
@@ -12,4 +12,29 @@ arguments:
         type:
             - int
             - binData
-            - array # of int
+        variadic: array
+tests:
+    -
+        name: 'Bit Position Array'
+        link: 'https://www.mongodb.com/docs/manual/reference/operator/query/bitsAllClear/#bit-position-array'
+        pipeline:
+            -
+                $match:
+                    a:
+                        $bitsAllClear: [1, 5]
+    -
+        name: 'Integer Bitmask'
+        link: 'https://www.mongodb.com/docs/manual/reference/operator/query/bitsAllClear/#integer-bitmask'
+        pipeline:
+            - $match:
+                    a:
+                        # $bitsAllClear: 35
+                        $bitsAllClear: [35]
+    -
+        name: 'BinData Bitmask'
+        link: 'https://www.mongodb.com/docs/manual/reference/operator/query/bitsAllClear/#bindata-bitmask'
+        pipeline:
+            - $match:
+                  a:
+                      $bitsAllClear:
+                          - { $binary: { base64: 'TUM9PQ==', subType: '00' } }
```

Comment:
I've no idea what I'm doing with this binary data. This is the Extended JSON representation of the PHP `new Binary('MC==')`, which I want to be the equivalent of the JS `BinData(0, "ID==")`

Reply:
Note: [`Binary::__construct()`](https://www.php.net/manual/en/mongodb-bson-binary.construct.php) takes a string argument representing binary data. So if you're constructing this in PHP, you should be doing `new Binary(base64_decode(...))`.

I have no idea why the docs use base64 `ID==` as the representation of binary `00100000`. Even the shell seems to disagree:

```
> BinData(0, "ID==")
Binary.createFromBase64("IA==", 0)
```

Using a tool like [aaronluna.dev/base64](https://aaronluna.dev/base64), you can see that `ID==` has the following representation:

```
ASCII
HEX     2   0
8-BIT   00100000

6-BIT   001000 000011
DECIMAL 8      3
BASE64  I      D     =     =
```

`IA==` is:

```
ASCII
HEX     2   0
8-BIT   00100000

6-BIT   001000 000000
DECIMAL 8      0
BASE64  I      D     =     =
```

That suggests `ID==` is non-canonical, since it denotes 12 bits of data. Only by truncating the last four bits do you end up with the same value as `IA==`. I think this is worth a [DOCSP](https://jira.mongodb.org/browse/DOCSP) ticket to suggest a change to the server docs.

libbson actually raises an error from [`_bson_json_parse_binary_elem()`](https://github.com/mongodb/mongo-c-driver/blob/1.25.4/src/libbson/src/bson/bson-json.c#L849) when parsing `ID==` (perhaps rightfully so):

>  Invalid input string "ID==", looking for base64-encoded binary

I'd suggest switching over to `IA==` in this example and leaving a comment in the YAML explaining that you modified the value from the docs to use the more canonical representation.

Reply:
Thanks, I'm fixing the BSON binary type in https://github.com/mongodb/mongo-php-builder/pull/34 using a Yaml tag.

The tag `!binary 'IA=='` will be converted to `new MongoDB\BSON\Binary(base64_decode('IA=='))`

Reply:
See [DOCSP-35733](https://jira.mongodb.org/browse/DOCSP-35733)